zio-cache/shared/src/main/scala/zio/cache/ManagedCache.scala

```diff
@@ -0,0 +1,87 @@
+package zio.cache
+
+import zio.{Exit, IO, Managed, TaskManaged, UIO, URIO}
+
+import java.time.Duration
+
+
+sealed abstract class ManagedCache[-Key, +Error, +Value] {
+
+  /**
+   * Returns statistics for this cache.
+   * @return
+   */
+  def cacheStats: UIO[CacheStats]
+
+  /**
+   * Return whether a Managed associated with the specified key exists in the cache.
+   * @param key
+   * @return
+   */
+  def contains(key: Key): UIO[Boolean]
+
+  /**
+   * Return statistics for the specified entry.
+   */
+  def entryStats(key: Key): UIO[Option[EntryStats]]
+
+  /**
+   * Give a proxy managed on the resource already created for this key if any.
+   * Otherwise computes the managed with the lookup function, puts it in the cache,
+   * and returns it.
+   * The given managed is lazy, before it's actually used the resource is not created and
+   * cached.
+   * The register creation time for the given resource is saved one the first managed used
+   * @param key
+   * @return
+   */
+  def get(key: Key): UIO[Managed[Error, Value]]
+
+  /**
+   * Force the reuse of the lookup function to compute the returned managed associated with the specified key,
+   * one first use of the given managed the new resource value if put in the cache and if any old resource it is cleaned
+   * (once all thread using it are done with it).
+   * Until the refresh actually produce a new resource, parallel call to get will still use the old resource
+   * @param key
+   * @return
+   */
+  def refresh(key: Key): UIO[Managed[Error, Value]]
```

Comment:
Not sure this is the best API that the refresh is really performed only once the given managed is actually used but this is what appear to be the more natural from an implementation perspective

Reply:
So you have in mind:
```
def refresh(key: Key): IO[Error, Unit]
```
That will force a non lazy recomputation of the resource?

Reply:
Now refresh do not lazily recomputed the resource.